### DISCUSSION TITLE
MDX

### DISCUSSION THREAD

**Discussion by mlajtos** [impulse-oss/impulse#7] [Q&A]
Looks really cool! Will it work with *.mdx i.e. Markdown+JSX (https://mdxjs.com) please?

**Comment by kirillrogovoy** [comment]
Hey Milan,

That's actually a great question. :) I've never checked.

I'll see if it works and what could be done if it doesn't.

**Reply by kirillrogovoy** [reply]
@mlajtos So, here are my findings:

1. Impulse still sees all the MDX components/elements in the navigation tree.
2. The source mapping from a `<div>` to your .mdx file is *not* preserved. It really depends on the MDX compiler, but in my tests, the mapping is just not there.

It means that you can still use Impulse for the app overall, but you won't be able to interactively edit anything inside your .mdx files, and jumping to the code would lead you to where you actually attempt to render the MDX content.

Not sure if there's much I can do, especially considering that there are many ways to use MDX (e.g. pre-compilation vs runtime compilation, etc.)
